alibi/prototypes/protoselect.py

```diff
@@ -0,0 +1,543 @@
+import os
+import logging
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.offsetbox import OffsetImage, AnnotationBbox
+
+from tqdm import tqdm
+from copy import deepcopy
+from typing import Callable, Optional, Dict, List, Union, Tuple
+from sklearn.model_selection import KFold
+from sklearn.neighbors import KNeighborsClassifier
+from skimage.transform import resize
+
+from alibi.utils.distance import batch_compute_kernel_matrix
+from alibi.api.interfaces import Explainer, Explanation, FitMixin
+from alibi.api.defaults import DEFAULT_META_PROTOSELECT, DEFAULT_DATA_PROTOSELECT
+from alibi.utils.kernel import EuclideanDistance
+
+logger = logging.getLogger(__name__)
+
+
+class ProtoSelect(Explainer, FitMixin):
```

Comment:
Let's discuss if this should even follow the `Explainer` interface...

Reply:
Decided to follow `Explainer` interface for now. To be changed in other iterations if required.